File: 1642-water-bottles/water-bottles.cpp
class Solution {
public:
    int numWaterBottles(int numBottles, int numExchange) {
        if(numExchange < 2) return -1;
        int cnt = 0, recycle = 0;
        while(numBottles >= numExchange){
            recycle = numBottles/numExchange;
            cnt += numExchange*recycle;
            numBottles = numBottles%numExchange + recycle;
        }
        cnt += numBottles;
        return cnt;
    }
};

File: 1642-water-bottles/water-bottles.py
class Solution:
    def numWaterBottles(self, numBottles: int, numExchange: int) -> int:
        if numExchange < 2: return -1
        cnt, bonus = 0,0
        while(numBottles >= numExchange):
            bonus = numBottles//numExchange
            cnt += bonus*numExchange
            numBottles = (numBottles % numExchange) + bonus 
        return cnt+numBottles